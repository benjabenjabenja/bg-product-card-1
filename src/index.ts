/**
 * ACA SE EXPORTAN TODOS LOS COMPONENTES, HOOKS, STYLES, Y LO QUE SEA QUE SE QUIERA EXPORTAR
 */

// COMPONENTES
export * from './components'

