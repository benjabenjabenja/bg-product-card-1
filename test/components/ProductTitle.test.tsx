import React from 'react';
import { render } from '@testing-library/react';
import { ProductTitle, ProductCard } from '../../src';
import { fakeMockProduct2 } from './data/products';


//const product = fakeMockProduct;
const product2 = fakeMockProduct2;
describe('ProductTitle', () => { 
    test('should render component with title personalized', () => {
        const { container } = render(<ProductTitle title="Test Product Title" className='test-class' />);
        expect(container).toMatchSnapshot();
    });

    test('should render component with title of product from product context', () => {
        const { container } = render(
            <ProductCard product={product2}>
                {
                    () => <ProductTitle />
                }
            </ProductCard>
        );
        
        expect(container).toMatchSnapshot();
    })
})
