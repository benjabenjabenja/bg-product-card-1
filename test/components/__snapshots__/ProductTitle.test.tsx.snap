// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should render component with title of product from product context 1`] = `
<div>
  <div
    class="productCard undefined"
  >
    <span
      class="productDescription undefined"
    >
      Product 2 - con imagen
    </span>
  </div>
</div>
`;

exports[`ProductTitle should render component with title personalized 1`] = `
<div>
  <span
    class="productDescription test-class"
  >
    Test Product Title
  </span>
</div>
`;
